### PR TITLE
lint: fix pre-commit warnings/errors

### PR DESCRIPTION
**Pull request by nilason**
Address pre-commit warnings and fixes. A majority of the issues originate from CI bot PRs, but not all.
This emphasises the need to explore the introduction of [pre-commit.ci](https://pre-commit.ci/).

### CODE REVIEW

File: .github/workflows/python-code-quality.yml [hunk 1]
```diff
@@ -114,7 +114,10 @@ jobs:
       - name: Run Pylint on other files using pytest
         run: |
           pip install pytest pytest-pylint==0.19
-          echo "::warning file=.github/workflows/python-code-quality.yml,line=116,col=42,endColumn=48::Temporarily downgraded pytest-pylint to allow merging other PRs. The errors reported with a newer version seem legitimite and should be fixed (2023-10-18, see https://github.com/OSGeo/grass/pull/3205)"
+          echo "::warning file=.github/workflows/python-code-quality.yml,line=116,col=42,endColumn=48::\
+            Temporarily downgraded pytest-pylint to allow merging other PRs. The errors reported\
+            with a newer version seem legitimite and should be fixed (2023-10-18,\
+            see https://github.com/OSGeo/grass/pull/3205)"
```
Comment (echoix) [inline .github/workflows/python-code-quality.yml:120]: It breaks the output of the warning, but that warning was written by me to intentionally mentioning that the test failures must be fixed one day to upgrade the package version.

Comment (echoix) [reply .github/workflows/python-code-quality.yml:120]: It can stay as is!

**Comment by echoix** [comment]
> The warning is there:
> 
> ```
> Successfully installed pytest-pylint-0.19.0
> Warning:   Temporarily downgraded pytest-pylint to allow merging other PRs. The errors reported  with a newer version seem legitimite and should be fixed (2023-10-18,  see https://github.com/OSGeo/grass/pull/3205)
> ============================= test session starts ==============================
> ```
> 
> with colours and all.

It is the extra spaces in between some words that come from the new indentation. But either way, the goal is to have it removed some day, so it doesn't matter at all.

**Comment by nilason** [comment]
> It is the extra spaces in between some words that come from the new indentation. But either way, the goal is to have it removed some day, so it doesn't matter at all.

Yes, there is one extra space after warning. But the warning is issued correctly. Multiline strings in yaml is a PITA.

**Comment by nilason** [comment]
I will update this after #3254 is merged.

**Pull request merged**